### PR TITLE
Initial validators ending e2e

### PR DESCRIPTION
**Pull request by iansuvak**
## Why this should be merged

Provides a pattern for calling `initializeValidatiorSet` as part of E2E testing. 

This will probably interact with the initial churn tracker stuff and we should make sure that the initial validators we are adding have reasonable weight. 

TODO:
- [x] Native 
- [x] ERC20
- [x] PoA->PoS case

## How this works
- References new AvaGo commit defining a new P-chain signable message 
- Adds manual packing for any subnetconversionData struct using reflect
- Adds helper functions in utils for PoA

## How this was tested
All E2E tests should pass except for 
Poa -> PoS conversion which just passes the step until PoS

All tests right now just add a single validator, we should add a test focusing on adding multiple and removing them. 
## How is this documented

### CODE REVIEW

**Review by geoff-vball — APPROVED** [review]
LGTM aside from Matt's comments

File: tests/flows/staking/poa_to_pos.go [hunk 1]
```diff
@@ -185,16 +197,7 @@ func PoAMigrationToPoS(network interfaces.LocalNetwork) {
 		stakeAmount,
 	)
 
-	// Attempt to delist previous PoA validator with wrong owner and check that it fails
```
Comment (iansuvak) [inline tests/flows/staking/poa_to_pos.go:188]: Anyone can delist PoA validators after conversion now so this test no longer applies

File: contracts/staking/ValidatorManager.sol [hunk 2]
```diff
@@ -150,9 +149,8 @@ abstract contract ValidatorManager is Initializable, ContextUpgradeable, IValida
             subnetConversionData.validatorManagerAddress,
             uint32(numInitialValidators)
         );
-
         uint256 totalWeight;
-        for (uint256 i; i < numInitialValidators; ++i) {
+        for (uint32 i; i < numInitialValidators; i++) {
```
Comment (minghinmatthewlam) [inline contracts/staking/ValidatorManager.sol:153]: think this was by accident for the i++ change

**Review by minghinmatthewlam — APPROVED** [review]
LGTM small increment variable change

**Pull request merged**